cpp/src/plasma/tf_plasma_op.cc

```diff
@@ -0,0 +1,383 @@
+      //                                                void {
+      //         int object_status;
+      //         {
+      //           mutex_lock lock(mu_);
+      //           LOG(INFO) << "Launching Info()";
+      //           ARROW_CHECK_OK(client_.Info(object_id, &object_status));
+      //           LOG(INFO) << "Done launching Info()";
+      //         }
+      //         CHECK(object_status != PLASMA_CLIENT_DOES_NOT_EXIST);
+
+      //         if (object_status == PLASMA_CLIENT_LOCAL) {
+      //           LOG(INFO) << "Object is local!";
+      //           CHECK(0);
+      //           // TODO(zongheng):  do real work;
+      //         } else {
+
+      //           LOG(INFO) << "Object still not local, status: " <<
+      //           object_status; std::function<void()> new_func =
+      //           WaitForTransfer; context->device()
+      //               ->tensorflow_gpu_device_info()
+      //               ->event_mgr->ThenExecute(h2d_stream, new_func);
+      //         }
+      //       };
+
+      //       context->device()->tensorflow_gpu_device_info()->event_mgr->ThenExecute(
+      //           h2d_stream, WaitForTransfer);
+
+      // Important.  See note in T2P op.
+      // We don't check the return status since the host memory might've been
+      // already registered (e.g., the TensorToPlasmaOp might've been run).
+      stream_executor->HostMemoryRegister(
+          const_cast<void*>(static_cast<const void*>(plasma_data)),
+          static_cast<uint64>(size_in_bytes));
+
+      perftools::gputools::DeviceMemoryBase wrapped_dst(
+          static_cast<void*>(output_tensor->flat<float>().data()));
+      const bool success =
+          h2d_stream
+              ->ThenMemcpy(&wrapped_dst, static_cast<const void*>(plasma_data),
+                           static_cast<uint64>(size_in_bytes))
+              .ok();
+      OP_REQUIRES_ASYNC(context, success,
+                        errors::Internal("H2D memcpy failed to be enqueued."), done);
+
+      // Without this sync the main compute stream might proceed to use the
+      // Tensor buffer, but its contents might still be in-flight from our
+      // h2d_stream.
+      CHECK(orig_stream->ThenWaitFor(h2d_stream).ok());
+
+      context->device()->tensorflow_gpu_device_info()->event_mgr->ThenExecute(
+          h2d_stream, std::move(done));
+#endif
+    }
+  }
+
+ private:
+  std::string plasma_store_socket_name_;
+  std::string plasma_manager_socket_name_;
+};
+
+REGISTER_OP("TensorToPlasma")
+    .Input("input_tensor: dtypes")
+    .Input("plasma_object_id: string")
+    .Attr("dtypes: list(type)")
+    .Attr("plasma_store_socket_name: string")
+    .Attr("plasma_manager_socket_name: string");
+
+REGISTER_KERNEL_BUILDER(Name("TensorToPlasma").Device(DEVICE_CPU),
+                        TensorToPlasmaOp<CPUDevice>);
+REGISTER_KERNEL_BUILDER(Name("TensorToPlasma").Device(DEVICE_GPU),
```

Comment:
Please wrap these two GPU registrations with GOOGLE_CUDA guard

Reply:
doing this